include/aikido/constraint/dart/IKRankingStrategy.hpp

```diff
@@ -0,0 +1,69 @@
+#ifndef AIKIDO_CONSTRAINT_DART_IKRANKINGSTRATEGY_HPP_
+#define AIKIDO_CONSTRAINT_DART_IKRANKINGSTRATEGY_HPP_
+
+#include <dart/dynamics/dynamics.hpp>
+#include "aikido/statespace/dart/MetaSkeletonStateSpace.hpp"
+
+namespace aikido {
+namespace constraint {
+namespace dart {
+
+class IKRankingStrategy
+{
+public:
+  // TODO (avk): Move this to protected?
+  // Read this in detail:
+  // 1. https://stackoverflow.com/questions/2290733/
+  // 1. initialize-parents-protected-members-with-initialization-list-c
+  // 2. https://stackoverflow.com/questions/18479295/
+  // 2. member-initializer-does-not-name-a-non-static-data-member-or-base-class
+  /// Constructor
+  ///
+  /// \param[in] metaSkeletonStateSpace Statespace of the skeleton.
+  /// \param[in] metaskeleton Metaskeleton of the robot.
+  /// \param[in] numIKSolution Number of IK solutions to rank.
+  IKRankingStrategy(
+      statespace::dart::ConstMetaSkeletonStateSpacePtr metaSkeletonStateSpace,
+      ::dart::dynamics::ConstMetaSkeletonPtr metaSkeleton,
```

Comment:
What would the `MetaSkeleton` be used for? If it's just to extract the current configuration, maybe this should be `currentState` instead?